internal/hns/hnsendpoint.go

```diff
@@ -76,6 +86,16 @@ type EndpointStats struct {
 // HNSEndpointRequest makes a HNS call to modify/query a network endpoint
 func HNSEndpointRequest(method, path, request string) (*HNSEndpoint, error) {
 	endpoint := &HNSEndpoint{}
+	if windowsBuild < osversion.RS5 {
+		switch method {
+		case "GET":
+			endpointMu.RLock()
+			defer endpointMu.RUnlock()
+		case "DELETE", "POST":
+			endpointMu.Lock()
+			defer endpointMu.Unlock()
+		}
+	}
```

Comment:
The only thing that comes to mind for this PR is a conundrum we ran into by calling osversion in another exposed API, namely that retrofitting these APIs with osversion means the caller now has to manifest to get the "new" behavior. The new behavior here if they don't manifest boils down to slower execution as we need to lock/unlock and they lose concurrent creates even if they're on a host that supports it, as osversion will return windows 8's build number if not manifested which will satisfy `if windowsBuild < osversion.RS5`.

For moby this doesn't matter as it's manifested and I'd say an overwhelming majority of uses of hcsshim is in containerd and moby, but something to think of. @jterry75 and I had played around with the thought of possibly having a fallback for osversion, where we'd parse `cmd /c ver`'s output if we find the program isn't manifested. That is a bit hacky though 😆